datarobot_batch_scoring/api_response_handlers/pred_api_v10.py

```diff
@@ -66,6 +67,16 @@ def format_data(result, batch, **opts):
             written_fields = out_fields[1:]
             comb = [row[1:] for row in pred]
 
+    if 'reasonCodes' in single_row:
+        num_reason_codes = len(single_row['reasonCodes'])
```

Comment:
Do we always have the same number of reasonCodes for all rows?

AFAIK we don't generate them for all rows and number of important features can be different for different rows..

And we don't have thresholds in request arguments - does it mean we calculate them for all rows or use default project settings?

Reply:
When the reason codes are requested, we generate reason codes for all fields, sort them, and then return only the requested number of them.